packages/muelu/src/Utils/MueLu_Monitor.hpp

```diff
@@ -149,10 +150,11 @@ namespace MueLu {
         @param[in] msg         String that indicates what the SubMonitor is monitoring, e.g., "Build"
         @param[in] msgLevel    Governs whether information should be printed.
         @param[in] timerLevel  Governs whether timing information should be *gathered*.  Setting this to NoTimeReport prevents the creation of timers.
+        @param[in] label       An optional prefix label.
     */
-    SubMonitor(const BaseClass& object, const std::string & msg, MsgType msgLevel = Runtime1, MsgType timerLevel = Timings1)
-      : printMonitor_(object, msg, msgLevel),
-        timerMonitor_(object, object.ShortClassName() + ": " + msg + " (sub, total)", timerLevel)
+    SubMonitor(const BaseClass& object, const std::string & msg, MsgType msgLevel = Runtime1, MsgType timerLevel = Timings1, std::string label = "")
+      : printMonitor_(object, label + msg, msgLevel),
+        timerMonitor_(object, label + object.ShortClassName() + ": " + msg + " (sub, total)", timerLevel)
```

Comment:
It seems like you are using "attach : to a label if it's not empty, otherwise do nothing" a lot. Would you mind creating a simple utility function for this matter? Or, if you prefer, when storing a label you could also store a label_colon.

Reply:
@aprokop Went for the helper function..